microimpute/comparisons/autoimpute.py

```diff
@@ -0,0 +1,191 @@
+            error_msg = f"Missing predictor columns in reciver data: {missing_predictors_receiver}"
+            log.error(error_msg)
+            raise ValueError(error_msg)
+
+        missing_imputed_donor = [
+            col for col in imputed_variables if col not in donor_data.columns
+        ]
+        if missing_imputed_donor:
+            error_msg = f"Missing imputed variable columns in donor data: {missing_imputed_donor}"
+            log.error(error_msg)
+            raise ValueError(error_msg)
+
+        # Validate that predictor columns exist in receiver data (imputed variables may not be present in receiver data)
+        missing_predictors_receiver = [
+            col for col in predictors if col not in receiver_data.columns
+        ]
+        if missing_predictors_receiver:
+            error_msg = f"Missing predictor columns in test data: {missing_predictors_receiver}"
+            log.error(error_msg)
+            raise ValueError(error_msg)
+
+        log.info(
+            f"Generating imputations to impute from {len(donor_data)} donor data to {len(receiver_data)} receiver data for variables {imputed_variables} with predictors {predictors}. "
+        )
+
+        # Step 1: Data preparation
+
+        ## Normalizing ?? should we drop the columns that are not predictors or imputed variables?
+
+        X_train, X_test, dummy_info = preprocess_data(
+            donor_data[predictors + imputed_variables],
+            train_size=train_size,
+            test_size=(1 - train_size),
+        )
+        receiver_data, dummy_info = preprocess_data(
+            receiver_data[predictors],
+            full_data=True,
+            train_size=train_size,
+            test_size=(1 - train_size),
+        )
+
+        if dummy_info:
+            # Retrieve new predictors and imputed variables after processed data
+            for orig_col, dummy_cols in dummy_info.items():
+                if orig_col in predictors:
+                    predictors.remove(orig_col)
+                    predictors + dummy_cols
+                elif orig_col in imputed_variables:
+                    imputed_variables.remove(orig_col)
+                    imputed_variables + dummy_cols
+
+        Y_test = X_test[imputed_variables]
+
+        # If imputed variables are in receiver data, remove them
+        receiver_data = receiver_data.drop(
+            columns=imputed_variables, errors="ignore"
+        )
+
+        # Step 2: Imputation with each method
+
+        ## We evaluate imputation methods with a test donor subset right? And we train on all passed data and hope the user makes decisions about size?
+
+        if not models:
+            # If no models are provided, use default models
+            model_classes: List[Type[Imputer]] = [QRF, OLS, QuantReg, Matching]
+
+        ## How do we want to handle the hyperparameters? Do we want to optimize in autoimpute as well and restructure all other functions to accept hyperparameters?
```

Comment:
Totally OK to separate the hyperparameter tuning out from this PR and file later- it's not critical that it's in here, though we do need to add it at some point.

Reply:
But yes tuning should go in this PR.